tests/ts_A_activate_table_pass_2.erl

```diff
@@ -0,0 +1,142 @@
+%% -------------------------------------------------------------------
+%%
+%% Copyright (c) 2015 Basho Technologies, Inc.
+%%
+%% This file is provided to you under the Apache License,
+%% Version 2.0 (the "License"); you may not use this file
+%% except in compliance with the License.  You may obtain
+%% a copy of the License at
+%%
+%%   http://www.apache.org/licenses/LICENSE-2.0
+%%
+%% Unless required by applicable law or agreed to in writing,
+%% software distributed under the License is distributed on an
+%% "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+%% KIND, either express or implied.  See the License for the
+%% specific language governing permissions and limitations
+%% under the License.
+%%
+%% -------------------------------------------------------------------
+
+-module(ts_A_activate_table_pass_2).
+
+-behavior(riak_test).
+
+-include_lib("eunit/include/eunit.hrl").
+
+-export([confirm/0]).
+
+confirm() ->
+    Cluster = ts_util:build_cluster(single),
+    % individual assert matches to show line numbers in failures
+    ?assertMatch(
+        {error_creating_bucket_type, _},
+        create_and_activate_bucket_type(Cluster, table_def("^"))
+    ),
+    ?assertMatch(
+        {error_creating_bucket_type, _},
+        create_and_activate_bucket_type(Cluster, table_def("$"))
+    ),
+    ?assertMatch(
+        {error_creating_bucket_type, _},
+        create_and_activate_bucket_type(Cluster, table_def("!"))
+    ),
+    ?assertMatch(
+        {error_creating_bucket_type, _},
+        create_and_activate_bucket_type(
+            Cluster, table_def("mytable", "!", "series", "time"))
+    ),
+    ?assertMatch(
+        {error_creating_bucket_type, _},
+        create_and_activate_bucket_type(
+            Cluster, table_def("mytable", "#@#@", "series", "time"))
+    ),
+    ?assertMatch(
+        {error_creating_bucket_type, _},
+        create_and_activate_bucket_type(
+            Cluster, table_def("mytable", "\\", "series", "time"))
+    ),
+    ?assertMatch(
+        {error_creating_bucket_type, _},
+        create_and_activate_bucket_type(
+            Cluster, table_def("mytable", "|", "series", "time"))
+    ),
+    ?assertMatch(
+        {error_creating_bucket_type, _},
+        create_and_activate_bucket_type(
+            Cluster, table_def("mytable", "?", "series", "time"))
+    ),
+    ?assertMatch(
+        {error_creating_bucket_type, _},
+        create_and_activate_bucket_type(
+            Cluster, table_def("mytable", "%", "series", "time"))
+    ),
+    % backticks need to be escaped or sh will error
+    ?assertMatch(
+        {error_creating_bucket_type, _},
+        create_and_activate_bucket_type(
+            Cluster, table_def("mytable", "\\`", "series", "time"))
```

Comment:
Ah, it becomes an arg to riak-admin and therefore, has to be esc-escaped. I stand corrected.

On December 10, 2015 4:59:25 PM GMT, Andy Till <notifications@github.com> wrote:
>> +    ),
>> +    ?assertMatch(
>> +        {error_creating_bucket_type, _},
>> +        create_and_activate_bucket_type(
>> +            Cluster, table_def("mytable", "?", "series", "time"))
>> +    ),
>> +    ?assertMatch(
>> +        {error_creating_bucket_type, _},
>> +        create_and_activate_bucket_type(
>> +            Cluster, table_def("mytable", "%", "series", "time"))
>> +    ),
>> +    % backticks need to be escaped or sh will error
>> +    ?assertMatch(
>> +        {error_creating_bucket_type, _},
>> +        create_and_activate_bucket_type(
>> +            Cluster, table_def("mytable", "\\`", "series", "time"))
>
>What did it give when you ran it without escapes?  For me it gives the
>output:
>
>```
>16:25:53.584 [info] Running:
>/home/vagrant/rt/riak/current/dev/dev1/bin/riak-admin "bucket-type"
>"create" "mytable" "{\"props\": {\"n_val\": 3, \"table_def\": \"CREATE
>TABLE mytable ( ` VARCHAR   NOT NULL, serie
>s VARCHAR   NOT NULL, time   TIMESTAMP NOT NULL, PRIMARY KEY ((`,
>series, quantum(time, 15, 'm')),  `, series, time))\"}}"
>16:25:54.586 [info] sh: 1: Syntax error: EOF in backquote substitution
>
>16:25:54.587 [info] Running:
>/home/vagrant/rt/riak/current/dev/dev1/bin/riak-admin "bucket-type"
>"activate" "mytable"
>16:25:55.589 [info] mytable is not an existing bucket type
>
>16:25:55.589 [error] Could not activate bucket type. Retrying. Result =
>{ok,"mytable is not an existing bucket type\n"}
>16:25:55.590 [error] Error in process <0.110.0> on node
>'riak_test@127.0.0.1' with exit value:
>{{assertMatch_failed,[{module,ts_A_activate_table_pass_2},{line,76},{expression,"create_and_activate_bucket_type
>( C
>luster , table_def ( \"mytable\" , \"`\" , \"series\" , \"time\" )
>)"},{pattern,"{ error_creating_b...
>```
>
>It fails because it is not valid shell script.
>
>---
>Reply to this email directly or view it on GitHub:
>https://github.com/basho/riak_test/pull/957/files#r47252909

-- 
Sent from my Android device with K-9 Mail. Please excuse my brevity.